Sources/Interaction/Widgets/ImageCroppingRegionsRepresentation/index.js

```diff
@@ -0,0 +1,390 @@
+        verts.set([slicePos + 0.01, yPLower, zBMax], 42);
+        verts.set([slicePos, yBMin, zBMax], 45);
+
+        centerVerts.set([slicePos, yPLower, zPLower], 0);
+        centerVerts.set([slicePos, yPUpper, zPLower], 3);
+        centerVerts.set([slicePos, yPUpper, zPUpper], 6);
+        centerVerts.set([slicePos, yPLower, zPUpper], 9);
+        break;
+      case Orientation.XZ:
+        verts.set([xBMin, slicePos, zBMin], 0);
+        verts.set([xPLower, slicePos, zBMin], 3);
+        verts.set([xPLower, slicePos + 0.01, zPLower], 6);
+        verts.set([xBMin, slicePos, zPLower], 9);
+        verts.set([xPUpper, slicePos + 0.01, zBMin], 12);
+        verts.set([xBMax, slicePos, zBMin], 15);
+        verts.set([xBMax, slicePos + 0.01, zPLower], 18);
+        verts.set([xPUpper, slicePos, zPLower], 21);
+        verts.set([xPUpper, slicePos + 0.01, zPUpper], 24);
+        verts.set([xBMax, slicePos, zPUpper], 27);
+        verts.set([xBMax, slicePos, zBMax], 30);
+        verts.set([xPUpper, slicePos, zBMax], 33);
+        verts.set([xBMin, slicePos, zPUpper], 36);
+        verts.set([xPLower, slicePos, zPUpper], 39);
+        verts.set([xPLower, slicePos + 0.01, zBMax], 42);
+        verts.set([xBMin, slicePos, zBMax], 45);
+
+        centerVerts.set([xPLower, slicePos, zPLower], 0);
+        centerVerts.set([xPUpper, slicePos, zPLower], 3);
+        centerVerts.set([xPUpper, slicePos, zPUpper], 6);
+        centerVerts.set([xPLower, slicePos, zPUpper], 9);
+        break;
+      case Orientation.XY:
+        verts.set([xBMin, yBMin, slicePos], 0);
+        verts.set([xPLower, yBMin, slicePos], 3);
+        verts.set([xPLower, yPLower, slicePos + 0.01], 6);
+        verts.set([xBMin, yPLower, slicePos], 9);
+        verts.set([xPUpper, yBMin, slicePos + 0.01], 12);
+        verts.set([xBMax, yBMin, slicePos], 15);
+        verts.set([xBMax, yPLower, slicePos + 0.01], 18);
+        verts.set([xPUpper, yPLower, slicePos], 21);
+        verts.set([xPUpper, yPUpper, slicePos + 0.01], 24);
+        verts.set([xBMax, yPUpper, slicePos], 27);
+        verts.set([xBMax, yBMax, slicePos], 30);
+        verts.set([xPUpper, yBMax, slicePos], 33);
+        verts.set([xBMin, yPUpper, slicePos + 0.01], 36);
+        verts.set([xPLower, yPUpper, slicePos], 39);
+        verts.set([xPLower, yBMax, slicePos + 0.01], 42);
+        verts.set([xBMin, yBMax, slicePos], 45);
+
+        centerVerts.set([xPLower, yPLower, slicePos], 0);
+        centerVerts.set([xPUpper, yPLower, slicePos], 3);
+        centerVerts.set([xPUpper, yPUpper, slicePos], 6);
+        centerVerts.set([xPLower, yPUpper, slicePos], 9);
+        break;
+      default:
+      // noop
+    }
+
+    // This requires updating whenever geometry updates
+    updateCenterOpacity();
+
+    model.regionPolyData.modified();
+    model.centerRegionPolyData.modified();
+
+    publicAPI.invokePlanesPositionChanged();
+    publicAPI.modified();
+  };
```

Comment:
Should we call modified() if the geometry did not actually changed?

Reply:
A call to updateGeometry always assumes the geometry has changed, since I didn't think it would matter if the geometry actually didn't change.